trio/_tests/test_util.py

```diff
@@ -170,13 +170,14 @@ def test_func(arg):
     assert test_func.__module__ == __name__
 
 
-def test_final_metaclass():
-    class FinalClass(metaclass=Final):
+def test_final_decorator() -> None:
+    @final
+    class FinalClass:
         pass
 
     with pytest.raises(TypeError):
 
-        class SubClass(FinalClass):
+        class SubClass(FinalClass):  # type: ignore[misc]
             pass
```

Comment:
Add a comment here explaining a bit more what's being tested (particularly that it's two tests in one, both a runtime check and a type-check), and that `misc` is for "inherit from final"